src/helpers.js

```diff
@@ -170,7 +170,7 @@ function getLockFilename( filename )
 {
   const name = filename.replace(/[^\w]+/g, '-');
 
-  return path.join( os.tmpdir(), `${name}.lock` );
+  return path.join( os.tmpdir(), `${name}-${process.pid}.lock` );
```

Comment:
Using names like this will not actually lock the resource (`assets-manifest.json` in this case) since there could be multiple lock files for the same resource and one process won't know if another process is using the file.

For example, lets say we have process `123` and `456`. If we use your proposed naming, we'd have `assets-manifest-json-123.lock` and `assets-manifest-json-456.lock`. Process `123` would not know anything about `assets-manifest-json-456.lock` and vice versa.

Reply:
> For example, lets say we have process 123 and 456. If we use your proposed naming, we'd have assets-manifest-json-123.lock and assets-manifest-json-456.lock. Process 123 would not know anything about assets-manifest-json-456.lock and vice versa.

This is the goal of the contribution, two different Webpack processes should not use the same lockfile because even if they both try to write `assets-manifest.json` they are in a different context so it does not concern the same file.

I'm not sure there is actually a scenario where you run 2 Webpack instances and write into the same output path without expecting conflicts :thinking: . If there is, it might be best to make the lock dependent  of the `compilation.compiler.outputPath` maybe by hashing it and putting the hash in the name?

Reply:
These lock files are already using the output path of the manifest.
https://github.com/webdeveric/webpack-assets-manifest/blob/60af212d948ed43301857b993f6c451a084abeeb/src/WebpackAssetsManifest.js#L420-L429

Reply:
https://github.com/webdeveric/webpack-assets-manifest/blob/60af212d948ed43301857b993f6c451a084abeeb/src/WebpackAssetsManifest.js#L420-L425

`output` is always set to `manifest.json` in my two processes even if in one process `compilation.compiler.outputPath` has for value `/foo` and in the other `/bar`.